official/mnist/mnist_eager.py

```diff
@@ -98,20 +100,25 @@ def test(model, dataset):
 def main(_):
   tfe.enable_eager_execution()
 
+  # Automatically determine device and data_format
   (device, data_format) = ('/gpu:0', 'channels_first')
   if FLAGS.no_gpu or tfe.num_gpus() <= 0:
     (device, data_format) = ('/cpu:0', 'channels_last')
+  # If data_format is defined in FLAGS, overwrite automatically set value.
+  if FLAGS.data_format is not None:
+    data_format = data_format
   print('Using device %s, and data format %s.' % (device, data_format))
 
   # Load the datasets
   train_ds = dataset.train(FLAGS.data_dir).shuffle(60000).batch(
-      FLAGS.batch_size)
+      FLAGS.batch_size).repeat(FLAGS.epochs_per_eval)
```

Comment:
Out of curiosity: Why is there an "epochs_per_eval"? Why does it make sense to go through the eval dataset more than once?

Reply:
This is the number of epochs to train between evaluations. 

In retrospect, the naming isn't the clearest. Would `train_eval_interval` make more sense (open to ideas)?

Reply:
Ah, sorry, somehow missed that this was `train_ds` and not `test_ds`. Anyway, yeah, is there a need to have this be larger than 1? Could we do away with the flag altogether? If not, maybe `--epochs_between_evals` or something?

Reply:
The flag can be disabled. The flag is mainly there for estimator training, where constantly switching between eval and train requires reloading/saving the last checkpoint is rather slow. The eager training loop doesn't have this issue, so I'll remove the flag for this file. 

I like the name `--epochs_between_eval`!